### PR TITLE
Develop Fill[SiliconFlow]

### PR DESCRIPTION
**Pull request by MARD1NO**
image

### CODE REVIEW

File: src/flag_gems/ops/fill.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    offset = pid * BLOCK_SIZE + cols
+    tl.store(out_ptr + offset, value, mask=offset < N)
+
+
+def fill(self, value):
+    logging.debug("GEMS FILL")
+    out = torch.empty_like(self)
+    N = out.numel()
+
+    BLOCK_SIZE = 128
```
Comment (StrongSpoon) [inline src/flag_gems/ops/fill.py]: what about setting block_size to 512?

Comment (MARD1NO) [reply src/flag_gems/ops/fill.py]: Fixed

File: src/flag_gems/__init__.py [hunk 2]
```diff
@@ -137,6 +137,7 @@ def enable(lib=aten_lib):
     # lib.impl("gather.out", gather_out, "CUDA")
     lib.impl("isclose", isclose, "CUDA")
     lib.impl("allclose", allclose, "CUDA")
+    lib.impl("fill.Scalar", fill, "CUDA")
```
Comment (Bowen12992) [inline src/flag_gems/__init__.py]: how about implement `fill.Tensor`as well?

**Review by Bowen12992 — COMMENTED** [review]
The performance pic seems broken... maybe we can update it or just use a markdown table

File: src/flag_gems/__init__.py (hunk 2 above)
Comment (MARD1NO) [reply src/flag_gems/__init__.py]: Done

File: src/flag_gems/ops/fill.py [hunk 3]
```diff
@@ -14,11 +14,17 @@ def fill_kernel(
     N,
     value,
     BLOCK_SIZE: tl.constexpr,
+    IS_VALUE_SCALAR: tl.constexpr,
```
Comment (Bowen12992) [inline src/flag_gems/ops/fill.py]: It's better to use two funcs

**Review by Bowen12992 — DISMISSED** [review]
LGTM

File: src/flag_gems/ops/fill.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+import torch
+import triton
+import triton.language as tl
+
+from ..utils import libentry
+
+
+@libentry()
+@triton.jit
```
Comment (StrongSpoon) [inline src/flag_gems/ops/fill.py]: recommend to register value_scalar as do_not_specialize

Comment (MARD1NO) [reply src/flag_gems/ops/fill.py]: Fixed

**Review by StrongSpoon — APPROVED** [review]
lgtm

**Pull request merged**